addons/test_mail_sms/tests/test_phone_blacklist.py

```diff
@@ -50,6 +50,32 @@ def test_phone_blacklist_internals(self):
             rec = self.env['mail.test.sms.bl'].search([('phone_sanitized_blacklisted', '=', True)])
             self.assertEqual(rec, self.env['mail.test.sms.bl'])
 
+    def test_phone_blacklist_unblacklisted(self):
+        """ This test check for scenario where user:
+            1. Blacklists a number  -> creating new active record)
+            2. Unblacklists it      -> making record unactive (archived)
+            3. Blacklists it again (by attempting to create new record)
+        Last step should just make existing record active again, and this test checks it.
+        """
+        phone_number = self.test_numbers[0]
+        bl_record = self.env['phone.blacklist'].sudo().create([{'number': phone_number, 'active': True}])
+        num_of_records = self.env['phone.blacklist'].with_context(active_test=False).search_count([])
+
+        bl_record.action_archive()
+
+        # Attempt to blacklist unblacklisted
+        self.env['phone.blacklist'].sudo().create([{'number': phone_number}])
+
+        self.assertTrue(bl_record.active, "Attempting to blacklist already-unblacklisted, should make the record active again")
+        self.assertEqual(num_of_records, self.env['phone.blacklist'].with_context(active_test=False).search_count([]),
+            "Number of records shouldn't change. (Records probably were recreated, instead of activated)")
+
+    def test_phone_create_unblacklisted(self):
```

Comment:
```suggestion
    def test_phone_blacklist_create_unblacklisted(self):
```

Reply:
+ beware alphabetical ordering (not just to annoy you, but tests are executed alphabetically so makes sense to order them following that order in the file, eases finding them when you have several of them failing)